common/rulesys.cpp

```diff
@@ -24,462 +24,451 @@
 	#include "ruletypes.h"
-	#define RULE_REAL(cat, rule, default_value, notes) \
-		{ #cat ":" #rule, Category__##cat, RealRule, Real__##rule, notes },
+	#define RULE_REAL(category_name, rule_name, default_value, notes) \
+		{ #category_name ":" #rule_name, Category__##category_name, RealRule, Real__##rule_name, notes },
 	#include "ruletypes.h"
-	#define RULE_BOOL(cat, rule, default_value, notes) \
-		{ #cat ":" #rule, Category__##cat, BoolRule, Bool__##rule, notes },
+	#define RULE_BOOL(category_name, rule_name, default_value, notes) \
+		{ #category_name ":" #rule_name, Category__##category_name, BoolRule, Bool__##rule_name, notes },
 	#include "ruletypes.h"
 	{ "Invalid Rule", _CatCount, IntRule }
 };
 
 RuleManager::RuleManager()
-:	m_activeRuleset(0),
-	m_activeName("default")
+	: m_activeRuleset(0),
+	  m_activeName("default")
 {
 	ResetRules(false);
 }
 
-RuleManager::CategoryType RuleManager::FindCategory(const char *catname) {
-	int i;
-	for (i = 0; i < _CatCount; i++) {
-		if (strcasecmp(catname, s_categoryNames[i]) == 0)
-			return((CategoryType)i);
+RuleManager::CategoryType RuleManager::FindCategory(std::string category_name)
+{
+	for (int i = 0; i < _CatCount; i++) {
+		if (Strings::Contains(category_name, s_categoryNames[i])) {
+			return static_cast<CategoryType>(i);
+		}
 	}
-	return(InvalidCategory);
+
+	return InvalidCategory;
 }
 
-bool RuleManager::ListRules(const char *catname, std::vector<const char *> &into) {
-	CategoryType cat = InvalidCategory;
-	if (catname != nullptr) {
-		cat = FindCategory(catname);
-		if (cat == InvalidCategory) {
-			Log(Logs::Detail, Logs::Rules, "Unable to find category '%s'", catname);
-			return(false);
+bool RuleManager::ListRules(std::string category_name, std::vector<std::string> &l)
+{
+	CategoryType category_type = InvalidCategory;
+	if (!category_name.empty()) {
+		category_type = FindCategory(category_name);
+		if (category_type == InvalidCategory) {
+			LogRulesDetail("Unable to find category '{}'.", category_name);
+			return false;
 		}
 	}
-	int i;
-	int rule_count = CountRules();
-	for (i = 0; i < rule_count; i++) {
-		const RuleInfo &rule = s_RuleInfo[i];
-		if (catname == nullptr || cat == rule.category) {
-			into.push_back(rule.name);
+
+	for (int i = 0; i < CountRules(); i++) {
+		const auto& r = s_RuleInfo[i];
+		if (category_name.empty() || category_type == r.category) {
+			l.push_back(r.name);
 		}
 	}
-	return(true);
+
+	return true;
 }
 
-bool RuleManager::ListCategories(std::vector<const char *> &into) {
-	int i;
-	for (i = 0; i < _CatCount; i++) {
-		into.push_back(s_categoryNames[i]);
+bool RuleManager::ListCategories(std::vector <std::string> &l)
+{
+	for (int i = 0; i < _CatCount; i++) {
+		l.push_back(s_categoryNames[i]);
 	}
-	return(true);
+
+	return true;
 }
 
-bool RuleManager::GetRule(const char *rule_name, std::string &return_value) {
+bool RuleManager::GetRule(std::string rule_name, std::string &rule_value)
```

Comment:
Probably want to pass strings by const ref (here and other functions) to avoid a lot of unnecessary copy construction